source/block-and-log-streamer.ts

```diff
@@ -1,186 +1,185 @@
+import { reconcileLogHistoryWithAddedBlock, reconcileLogHistoryWithRemovedBlock } from './log-reconciler'
 
-import { List as ImmutableList } from "immutable";
-import * as createUuid from "uuid";
+import { List as ImmutableList } from 'immutable'
+import * as createUuid from 'uuid'
 
 export interface Configuration {
 	/** number of blocks to retain in history, defaults to 100 */
 	blockRetention?: number
 }
 
 export class BlockAndLogStreamer<TBlock extends Block, TLog extends Log> {
-	private lastKnownGoodBlockHistory: BlockHistory<TBlock> = ImmutableList<TBlock>();
-	private blockHistory: Promise<BlockHistory<TBlock>> = Promise.resolve(this.lastKnownGoodBlockHistory);
-	private lastKnownGoodLogHistory: LogHistory<TLog> = ImmutableList<TLog>();
-	private logHistory: Promise<LogHistory<TLog>> = Promise.resolve(this.lastKnownGoodLogHistory);
-	private pendingCallbacks: Array<() => void> = [];
+	private lastKnownGoodBlockHistory: BlockHistory<TBlock> = ImmutableList<TBlock>()
+	private blockHistory: Promise<BlockHistory<TBlock>> = Promise.resolve(this.lastKnownGoodBlockHistory)
+	private lastKnownGoodLogHistory: LogHistory<TLog> = ImmutableList<TLog>()
+	private logHistory: Promise<LogHistory<TLog>> = Promise.resolve(this.lastKnownGoodLogHistory)
+	private pendingCallbacks: Array<() => void> = []
 
-	private readonly blockRetention: number;
+	private readonly blockRetention: number
 
-	private readonly getBlockByHash: (hash: string) => Promise<TBlock | null>;
-	private readonly getLogs: (filterOptions: FilterOptions) => Promise<TLog[]>;
-	private readonly onError: (error: Error) => void = () => {};
+	private readonly getBlockByHash: (hash: string) => Promise<TBlock | null>
+	private readonly getLogs: (filterOptions: FilterOptions) => Promise<TLog[]>
+	private readonly onError: (error: Error) => void = () => {}
 
 	private readonly logFilters: { [propName: string]: Filter } = {}
-	private readonly onBlockAddedSubscribers: { [propName: string]: (block: TBlock) => void } = {};
-	private readonly onBlockRemovedSubscribers: { [propName: string]: (block: TBlock) => void } = {};
-	private readonly onLogAddedSubscribers: { [propName: string]: (log: TLog) => void } = {};
-	private readonly onLogRemovedSubscribers: { [propName: string]: (log: TLog) => void } = {};
+	private readonly onBlockAddedSubscribers: {
+		[propName: string]: (block: TBlock) => void
+	} = {}
+	private readonly onBlockRemovedSubscribers: {
+		[propName: string]: (block: TBlock) => void
+	} = {}
+	private readonly onLogAddedSubscribers: {
+		[propName: string]: (log: TLog) => void
+	} = {}
+	private readonly onLogRemovedSubscribers: {
+		[propName: string]: (log: TLog) => void
+	} = {}
 
 	/**
-	 * @param getBlockByHash async function that returns a block given a particular hash or null/throws if the block is not found
+	 * @param getBlockByHash async function that returns a block given a particular hash or null/throws if the block
+	 * is not found
```

Comment:
Prose text _definitely_ should be soft-wrapped, not word wrapped (I take a pretty hard-line stance on this one, though I'm open to discussing arguing me out of it will be incredibly hard).

Reply:
done.